### PR TITLE
Replace `.format` with f-string

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Replace `.format` with f-string. I ran `ruff --select UP032,RUF010 .` and made a few changes manually.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/tracking/default_experiment/registry.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def register_entrypoints(self):
             except (AttributeError, ImportError) as exc:
                 warnings.warn(
                     "Failure attempting to register default experiment"
-                    + 'context provider "{}": {}'.format(entrypoint.name, str(exc)),
+                    + f'context provider "{entrypoint.name}": {exc}',
```
Comment (gabrielfu) [inline mlflow/tracking/default_experiment/registry.py:50]: is `+` redundant?

File: tests/deployments/test_cli.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ def test_run_local():
     )
     assert f"Deployed locally at the key {f_name}" in res.stdout
     assert f"using the model from {f_model_uri}." in res.stdout
-    assert "It's flavor is {} and config is {}".format(f_flavor, str({})) in res.stdout
+    assert f"It's flavor is {f_flavor} and config is {str({})}" in res.stdout
```
Comment (gabrielfu) [inline tests/deployments/test_cli.py:127]:
```suggestion
    assert f"It's flavor is {f_flavor} and config is {{}}" in res.stdout
```

Comment (harupy) [reply tests/deployments/test_cli.py:127]: Thanks!

Comment (harupy) [reply tests/deployments/test_cli.py:127]: I don't want to rerun CI again. We can fix this in a follow-up.

File: mlflow/tracking/default_experiment/registry.py (hunk 1 above)
Comment (harupy) [reply mlflow/tracking/default_experiment/registry.py:50]: Yeah, I'll remove this in a follow-up.

Comment (harupy) [reply mlflow/tracking/default_experiment/registry.py:50]: @gabrielfu Ruff has https://beta.ruff.rs/docs/rules/explicit-string-concatenation. Do you want to enable this?

**Comment by harupy** [comment]
The `sentence_transformers` failure is unrelated. I already fixed it.

File: tests/deployments/test_cli.py (hunk 2 above)
Comment (gabrielfu) [reply tests/deployments/test_cli.py:127]: sure!

**Review by gabrielfu — APPROVED** [review]
LGTM!

**Pull request merged**